specifyweb/frontend/js_src/lib/components/SchemaViewer/TableList.tsx

```diff
@@ -117,22 +118,27 @@ export function GenericSortedDataViewer<
           const children = Object.keys(headers).map((column) => {
             const data = row[column];
             return (
-              <Cell
+              <div
                 className={cellClassName?.(row, column, index)}
                 key={column}
+                role="cell"
               >
                 {Array.isArray(data) ? data[1] : row[column]}
-              </Cell>
+              </div>
             );
           });
-          const key = `${row[indexColumn]?.toString()}-${index}`;
+          const key = row[indexColumn];
+          if (process.env.NODE_ENV === 'development' && typeof key !== 'string')
+            throw new Error(
+              `Expected index column to be string. Instead found: ${typeof key}`
+            );
           const link = getLink?.(row);
           return typeof link === 'string' ? (
-            <Link.Default href={link} key={key} role="row">
+            <Link.Default href={link} key={key as string} role="row">
```

Comment:
(optional) .toString() instead of `as string`?
if `key` is somehow an object, both cases would likely be a bit broken, but at least you avoid a type assertion

Reply:
Hmm toString() still might return undefined so I'd need to have a type assertion there too